src/SonarScanner.MSBuild.Shim/SonarScanner.Wrapper.cs

```diff
@@ -93,11 +93,11 @@ private static bool InternalExecute(AnalysisConfig config, IEnumerable<string> u
             return ExecuteJavaRunner(config, userCmdLineArguments, logger, exeFileName, fullPropertiesFilePath, new ProcessRunner(logger));
         }
 
-        internal /* for testing */ static string FindScannerExe(IPlatformHelper platformHelper = null)
+        internal /* for testing */ static string FindScannerExe(IOperatingSystemProvider operatingSystemProvider = null)
         {
-            platformHelper ??= EnvironmentBasedPlatformHelper.Instance;
+            operatingSystemProvider ??= OperatingSystemProvider.Instance;
             var binFolder = Path.GetDirectoryName(typeof(SonarScannerWrapper).Assembly.Location);
-            var fileExtension = platformHelper.OperatingSystem == PlatformOS.Windows ? ".bat" : string.Empty;
+            var fileExtension = operatingSystemProvider.OperatingSystem == PlatformOS.Windows ? ".bat" : string.Empty;
```

Comment:
It looks to me that this check is not related to the JRE download. It just needs to know if it runs on Windows or not to escape the command line arguments. I would not change the current logic.

Reply:
Indeed. There is the scanner wrapper, the TFS wrapper, and the MsBuildPathSettings. I think it would make sense to change all of them but the rework is not trivial. In all these places, we would need access to the aggregated settings, and in the end step, we don't have access to the `ProcessedArgs`, so things need to be moved around. 

Since this is not a requirement for the current sprint, I would create a separate issue and tackle it in a different sprint.